tsdb/engine/b1/b1.go

```diff
@@ -534,6 +534,9 @@ func (tx *Tx) Cursor(key string) tsdb.Cursor {
 	// Retrieve key bucket.
 	b := tx.Bucket([]byte(key))
 
+	tx.engine.mu.Lock()
```

Comment:
Should we also take the lock on entry first into the function? It's not a big deal, but I, for one, like to see function-level locking like this done as the first thing in the function (unless there is a reason now too) as it is makes the code a little clear to follow.